system/Session/Handlers/FileHandler.php

```diff
@@ -69,7 +69,7 @@ public function __construct(AppConfig $config, string $ipAddress)
 
         if (! empty($this->savePath)) {
             $this->savePath = rtrim($this->savePath, '/\\');
-            ini_set('session.save_path', $this->savePath);
+            if(!headers_sent()) ini_set('session.save_path', $this->savePath);
```

Comment:
Thank you for trying to improve CI4!

But the change seems to make the session not work correctly.
If the `if` condition is false, the session save path is not set correctly.

Why do you need this patch?

Reply:
I recently tried [gitpod.io](//gitpod.io) and my site was producing an error, `exception ini_set(): session ini settings cannot be changed after headers have already been sent` so I made those few changes to make it work. I'm thinking it might be the same issue as on #7441

Reply:
I don't know gitpod well, but in my understanding it provides development environment (IDE).
I tried it with my sample repository and the app works well without errors.

Please try your site with `php spark serve` on your local PC.

Reply:
Okay thanks. I have just tested this repository as well and there are no error. I'm only seeing this error on gitpod and not on my local PC so I'll close this pull request. Thanks a lot

Reply:
Thanks for your feedback.

If you send a PR next time, please sign your Git commits with GPG.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#signing
and you can fix the coding style with one command.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#php-style

Reply:
I apologize, I had not read the guidelines. Will do so next time.